jetty-http2/http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/HpackEncoder.java

```diff
@@ -342,12 +342,12 @@ public void encode(ByteBuffer buffer, HttpField field)
                     if (_debug)
                         encoding = indexed ? "PreEncodedIdx" : "PreEncoded";
                 }
-                // has the custom header name been seen before?
-                else if (name == null)
+                // has the custom header name been seen before and will it fit in dynamic table?
```

Comment:
The comment is wrong, should be "the custom header is unknown". I would just remove the comment as the code is clear and the comment inside the branch repeats the logic.

Reply:
The comment is not wrong.  The whole header has not been seen, but the name itself has been (but with a different value).